static/js/components/EducationTab.js

```diff
@@ -0,0 +1,240 @@
+    updateProfile: React.PropTypes.func,
+    errors: React.PropTypes.object,
+    dispatch: React.PropTypes.func.isRequired
+  };
+
+  static defaultProps = {
+    requiredFields: [],
+    validationMessages: {
+      'degree_name': 'Degree level',
+      'graduation_date': 'Graduation Date',
+      'field_of_study': 'Field of study',
+      'online_degree': 'Online Degree',
+      'school_name': 'School name',
+      'school_city': 'City',
+      'school_state_or_territory': 'State',
+      'school_country': 'Country'
+    }
+  };
+  static nestedValidationKeys = [
+    'degree_name',
+    'graduation_date',
+    'field_of_study',
+    'online_degree',
+    'school_name',
+    'school_city',
+    'school_state_or_territory',
+    'school_country'
+  ];
+  static validation(profile, requiredFields) {
+
+    let nestedFields = (index) => {
+      let keySet = (key) => ['education', index, key];
+      return EducationTab.nestedValidationKeys.map(key => keySet(key));
+    };
+    return requiredFields.concat(
+      ...profile.education.map((v, i) => nestedFields(i))
+    );
+  }
+
+  openNewEducationForm(level, index) {
+    const {dispatch, profile, updateProfile} = this.props;
+    let new_index = index;
+    if (index === null){
+      new_index = profile['education'].length;
+    }
+    /* add empty education */
+    let clone = Object.assign({}, profile);
+    clone['education'] = clone['education'].concat(generateNewEducation(level));
+    updateProfile(clone);
+    dispatch(openEducationForm(level, new_index));
+  }
+
+  printAddDegree(level){
+    const { educationLevels } = this.props;
+    if (educationLevels[level.value]){
+      return <Grid key={"add-"+level.value}>
+        <Cell col={11}></Cell>
+        <Cell col={1}>
+          <FABButton mini onClick={this.openNewEducationForm.bind(this, level.value, null )} raised ripple>
+            <Icon name="add" />
+          </FABButton>
+        </Cell>
+      </Grid>;
+    }
+    return null;
+  }
+
+  printExistingEducations(level) {
+    const {profile} = this.props;
+    return profile['education'].map(education => {
+      if (education.degree_name === level.value && 'id' in education) {
+        return <Grid className="existing-education-grid">
+          <Cell col={3}>{this.educationLevelLabels[education.degree_name]}</Cell>
+          <Cell col={7}>{education.graduation_date}</Cell>
+          <Cell col={1}>
+            <Button onClick={this.openEditEducationForm.bind(this, education.degree_name, education.id)}>
+              <Icon name="edit"/>
+            </Button>
+          </Cell>
+          <Cell col={1}>
+            <Button ><Icon name="delete"/></Button>
+          </Cell>
+        </Grid>;
+      }
+    });
+  }
+
+  openEditEducationForm(level, education_id){
+    const { dispatch, profile } = this.props;
+
+    let index = profile['education'].findIndex((education) => {
+      return ('id' in education) && education_id === education.id;
```

Comment:
We shouldn't need the `('id' in education)` part here. `education_id` will always be a number, so the `education_id === education.id` comparison should only be true if `education.id` is also a number